src/dynamic_boundary_conditions/tide/main_tide_slr.py

```diff
@@ -26,6 +28,161 @@
+        Additional options for medium confidence are: 'SSP1-1.9' or 'SSP3-7.0'.
+    add_vlm : bool
+        Indicates whether Vertical Land Motion (VLM) should be included in the sea level rise data.
+        Set to True if VLM should be included, False otherwise.
+    percentile : int
+        The desired percentile for the sea level rise data. Valid values are 17, 50, or 83.
+    increment_year : int = 1
+
+    Returns
+    -------
+    ValidationResult
+        Result of the validation, with validation failure reason if applicable.
+    """
+    valid_parameters = get_valid_parameters_based_on_confidence_level()
+
+    # Check if the provided confidence level is valid
+    if confidence_level not in valid_parameters:
+        return ValidationResult(False,
+                                f"Invalid value '{confidence_level}' for confidence_level. "
+                                f"Must be one of {list(valid_parameters.keys())}.")
+
+    # Check if the provided SSP scenario is valid
+    valid_ssp_scenarios = valid_parameters[confidence_level]['ssp_scenarios']
+    if ssp_scenario not in valid_ssp_scenarios:
+        return ValidationResult(False,
+                                f"Invalid value '{ssp_scenario}' for ssp_scenario."
+                                f" Must be one of {valid_ssp_scenarios}.")
+
+    # Check if the provided add_vlm value is valid
+    valid_add_vlms = [True, False]
+    if add_vlm not in valid_add_vlms:
+        return ValidationResult(False,
+                                f"Invalid value '{add_vlm}' for add_vlm. Must be one of {valid_add_vlms}.")
+
+    # Get the percentiles from the column names and check if the provided percentile value is valid
+    column_names_query = text(r"""
+        SELECT column_name
+        FROM information_schema.columns
+        WHERE
+            table_name = 'sea_level_rise'
+            AND table_catalog=:db_name
+            AND column_name ~ '^p\d+'
+        """).bindparams(db_name=config.get_env_variable("POSTGRES_DB"))
+    percentile_col_tuples = engine.execute(column_names_query).fetchall()
+    # Flatten the result of the query
+    percentile_cols = [col_tuple[0] for col_tuple in percentile_col_tuples]
+    # Remove the leading 'p'
+    valid_percentiles = [int(col[1:]) for col in percentile_cols]
+
+    if percentile not in valid_percentiles:
+        return ValidationResult(
+            False,
+            f"Invalid value '{percentile}' for percentile. Must be one of {valid_percentiles}.")
+
+    # Check if the provided projection year is valid
+    min_year = valid_parameters[confidence_level]['min_year']
+    max_year = valid_parameters[confidence_level]['max_year']
+    valid_proj_year = range(min_year, max_year + increment_year, increment_year)
+    if proj_year not in valid_proj_year:
+        return ValidationResult(
+            False,
+            f"Invalid value '{proj_year}' for proj_year. Must be one of {list(valid_proj_year)}.")
```

Comment:
I understand I think, since I originally had this set for the same year. However you pushed back on it last time and it appears some other validation code requires a future year.

I think it would be better to allow the user more choice and allow the current year. Is it okay if I try that and test it?

Reply:
I believe this was always set to `2030` in the main() function, and last time, I tried testing it using years prior to 2025, it was causing problems. Even if you try running the code with 2024 now it would not run, you'd still get that invalid value error. This is because it is dependent on the king tide data at the time that's fetched. I can explain over teams meeting if needed.

Reply:
I may need a discussion about this a little later, I'll send you a message after my next meeting